wom/src/test/scala/wom/types/WomFileTypeSpec.scala

```diff
@@ -0,0 +1,124 @@
+package wom.types
+
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.scalatest.{FlatSpec, Matchers}
+import spray.json.{JsNumber, JsString}
+import wom.values.{WomFloat, WomGlobFile, WomSingleDirectory, WomSingleFile, WomString}
+
+import scala.util.Success
+
+class WomFileTypeSpec extends FlatSpec with Matchers with TableDrivenPropertyChecks {
+
+  behavior of "WomFileType"
+
+  lazy val coercionTests = Table(
+    ("description", "womFileType", "value", "expected"),
+
+    ("a string to a dir", WomSingleDirectoryType, "example/string", WomSingleDirectory("example/string")),
+    ("a string to a file", WomSingleFileType, "example/string", WomSingleFile("example/string")),
+    ("a string to a glob", WomGlobFileType, "example/string", WomGlobFile("example/string")),
+
+    ("a js string to a dir", WomSingleDirectoryType, JsString("example/js"), WomSingleDirectory("example/js")),
+    ("a js string to a file", WomSingleFileType, JsString("example/js"), WomSingleFile("example/js")),
+    ("a js string to a glob", WomGlobFileType, JsString("example/js"), WomGlobFile("example/js")),
+
+    ("a wom string to a dir", WomSingleDirectoryType, WomString("example/wom"), WomSingleDirectory("example/wom")),
+    ("a wom string to a file", WomSingleFileType, WomString("example/wom"), WomSingleFile("example/wom")),
+    ("a wom string to a glob", WomGlobFileType, WomString("example/wom"), WomGlobFile("example/wom")),
+
+    ("a wom dir to a dir", WomSingleDirectoryType, WomSingleDirectory("example/dir"),
+      WomSingleDirectory("example/dir")),
+    ("a wom file to a file", WomSingleFileType, WomSingleFile("example/dir"), WomSingleFile("example/dir")),
+    ("a wom glob to a glob", WomGlobFileType, WomGlobFile("example/glob*"), WomGlobFile("example/glob*"))
+  )
+
+  forAll(coercionTests) { (description, womFileType, value, expected) =>
```

Comment:
An observation: you could remove the `womFileType` field and just use `expected.womType` to make these test cases a little shorter